src/models/jobProcessor.ts

```diff
@@ -49,35 +50,50 @@ export class JobProcessor {
   private async consumeAndProcess(): Promise<void> {
     const logCtx: LogContext = { ...this.logContext, function: this.consumeAndProcess.name };
     try {
-      const job = await this.getJob();
+      const jobAndTaskType = await this.getJobWithTaskType();
 
-      if (!job) {
+      if (!jobAndTaskType) {
         await setTimeoutPromise(this.dequeueIntervalMs);
         return;
       }
 
-      const jobHandler = this.jobHandlerFactory(job.type);
-      await jobHandler.handleJob(job);
+      await this.processJob(jobAndTaskType);
     } catch (error) {
       this.logger.error({ msg: 'Failed processing the job', error, logContext: logCtx });
       await setTimeoutPromise(this.dequeueIntervalMs);
     }
   }
 
-  private async getJob(): Promise<IJobResponse<unknown, unknown> | undefined> {
-    const logCtx: LogContext = { ...this.logContext, function: this.getJob.name };
-    for (const jobType of this.jobTypes) {
-      this.logger.debug({ msg: `try to dequeue task of type "${this.initTaskType}" and job of type "${jobType}"`, logContext: logCtx });
-      const task = await this.queueClient.dequeue(jobType, this.initTaskType);
+  private async processJob(jobAndTaskType: { job: IJobResponse<unknown, unknown>; taskType: string }): Promise<void> {
```

Comment:
better define it as type/interface:  `{ job: IJobResponse<unknown, unknown>; taskType: string }` its also being used by the response type of 'getJobWithTaskType()':
`private async getJobWithTaskType(): Promise<{ job: IJobResponse<unknown, unknown>; taskType: string } | undefined>`

Reply:
extract to new JobAndTaskType interface